api/src/main/java/org/pmiops/workbench/conceptset/mapper/ConceptSetMapper.java

```diff
@@ -18,4 +34,80 @@
+  @Mapping(target = "description", source = "conceptSet.description")
+  @Mapping(
+      target = "creationTime",
+      source = "conceptSet.creationTime",
+      qualifiedByName = "toTimestampCurrentIfNull")
+  @Mapping(
+      target = "lastModifiedTime",
+      source = "conceptSet.lastModifiedTime",
+      qualifiedByName = "toTimestampCurrentIfNull")
+  @Mapping(target = "participantCount", source = "conceptSet.participantCount")
+  @Mapping(target = "domain", ignore = true)
+  @Mapping(target = "survey", ignore = true)
+  @Mapping(target = "creator", ignore = true)
+  @Mapping(target = "surveysEnum", source = "conceptSet.survey")
+  @Mapping(target = "version", source = "conceptSet.etag", qualifiedByName = "etagToCdrVersion")
+  @Mapping(target = "workspaceId", ignore = true)
+  @Mapping(target = "conceptIds", ignore = true)
+  DbConceptSet clientToDbModel(
+      CreateConceptSetRequest source,
+      @Context Long workspaceId,
+      @Context DbUser creator,
+      @Context ConceptService conceptService,
+      @Context ConceptBigQueryService conceptBigQueryService);
+
+  @AfterMapping
+  default void addConceptIdsToSet(
+      CreateConceptSetRequest source,
+      @Context Long workspaceId,
+      @Context DbUser creator,
+      @MappingTarget DbConceptSet dbConceptSet,
+      @Context ConceptService conceptService,
+      @Context ConceptBigQueryService conceptBigQueryService) {
+    dbConceptSet.setWorkspaceId(workspaceId);
+    dbConceptSet.setCreator(creator);
+    Domain domainEnum = dbConceptSet.getDomainEnum();
+    if (domainEnum == null) {
+      throw new BadRequestException(
+          "Domain " + source.getConceptSet().getDomain() + " is not allowed for concept sets");
+    }
+    Iterable<DbConcept> concepts = conceptService.findAll(source.getAddedIds());
+    if (dbConceptSet.getConceptIds().size() > 1000) {
+      throw new BadRequestException("Exceeded 1000 in concept set");
+    }
+    List<DbConcept> mismatchedConcepts =
+        ImmutableList.copyOf(concepts).stream()
+            .filter(concept -> !concept.getConceptClassId().equals("Question"))
+            .filter(
+                concept -> {
+                  Domain domain =
+                      Domain.PHYSICALMEASUREMENT.equals(domainEnum)
+                          ? Domain.PHYSICALMEASUREMENT
+                          : DbStorageEnums.domainIdToDomain(concept.getDomainId());
+                  return !domainEnum.equals(domain);
+                })
+            .collect(Collectors.toList());
+    if (!mismatchedConcepts.isEmpty()) {
+      String mismatchedConceptIds =
+          Joiner.on(", ")
+              .join(
+                  mismatchedConcepts.stream()
+                      .map(DbConcept::getConceptId)
+                      .collect(Collectors.toList()));
+      throw new BadRequestException(
+          String.format("Concepts [%s] are not in domain %s", mismatchedConceptIds, domainEnum));
+    }
```

Comment:
Not sure any of this code is necessary. A `CreateConceptSetRequest` has a `ConceptSet` which can have only 1 domain. If we really feel like we need to validate that the concept ids all belong in the same domain, then we should check this in the controller. Also, there is a much easier way to validate this, just query the concept table for domain count.